### PR TITLE
Fix target block on fast sync restart

### PR DESCRIPTION
**Pull request by LukasGasior1**

### CODE REVIEW

File: src/main/scala/io/iohk/ethereum/blockchain/sync/SyncController.scala [hunk 1]
```diff
@@ -267,27 +273,30 @@ class SyncController(
...
       syncStatePersistCancellable.cancel()
+      syncStateStorageActor ! PoisonPill
+
       fastSyncStateStorage.purge()
-      log.info("Fast sync finished")
       appStateStorage.fastSyncDone()
-      context.parent ! FastSyncDone
+
+      log.info("Fast sync finished")
+      context.parent ! FastSyncDone // TODO: is this still needed (seems to be used for BlockBroadcastActor)
```
Comment (AlanVerbner) [inline src/main/scala/io/iohk/ethereum/blockchain/sync/SyncController.scala]: I guess that actor is going to be removed, isn't it?

Comment (ntallar) [inline src/main/scala/io/iohk/ethereum/blockchain/sync/SyncController.scala]: Yes, this was removed in https://github.com/input-output-hk/etc-client/pull/91 as they are no longer used

**Comment by rtkaczyk** [comment]
Could you provide some description? I don't fully understand the problem

**Comment by LukasGasior1** [comment]
Sure @rtkaczyk. When resuming fast sync after app restart we didn't use the same target block as before, instead we were choosing another one (which lead to invalid mpt state download).

**Pull request merged**